core/src/apps/cardano/shelley/address.py

```diff
@@ -0,0 +1,53 @@
+from trezor.crypto import bech32, hashlib
+
+from apps.common import HARDENED
+from apps.common.seed import remove_ed25519_prefix
+from apps.cardano.shelley import CURVE
+from apps.common import paths
+
+
+def validate_full_path(path: list) -> bool:
+    """
+    Validates derivation path to fit 1852'/1815'/a'/{0,1}/i,
+    where `a` is an account number and i an address index.
+    The max value for `a` is 20, 1 000 000 for `i`.
+    """
+    if len(path) != 5:
+        return False
+    if path[0] != 1852 | HARDENED:
+        return False
+    if path[1] != 1815 | HARDENED:
+        return False
+    if path[2] < HARDENED or path[2] > 20 | HARDENED:  # TODO do we still limit it to 20 accounts?
+        return False
+    if path[3] != 0 and path[3] != 1:
```

Comment:
Shouldn't this also allow a value of 2 for staking paths? I suppose it is not absolutely clear right now, but certificate paths might also need to be validated. Maybe adding a todo, so we don't forget, would be enough for now.